packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1117,6 +1145,16 @@ export class ContainerRuntime
 		this.heuristicsDisabled = this.isHeuristicsDisabled();
 		this.maxOpsSinceLastSummary = this.getMaxOpsSinceLastSummary();
 		this.initialSummarizerDelayMs = this.getInitialSummarizerDelayMs();
+		if (this.runtimeOptions.enableRuntimeIdCompressor) {
```

Comment:
Couple of questions:
1. Don't all clients connected to the same document have to have id compression enabled for it to work? What happens if some clients have it enabled and others don't?
2. Are you planning to support for old documents as well? Basically, documents that open with the version of container runtime that has this enabled. If yes, then the same question as #1 applies here. How will you ensure that all clients have this enabled?

Reply:
All clients will need to have this enable. I'm not sure I see a better way of handling this than making sure we hit a "saturation" point with regard to version and then explicitly error if the compressor isn't enabled and you receive an IdAllocationOp.

Reply:
Even if you reach saturation, anyone could disable this option. Imagine a loop container is created using `loop-page-container` which has this setting enabled. This container is openedin another app which loads another container (say `office-container`) which has this setting disabled. You have a problem now.

For features like this that require every container instance to be on the same page, you could do the following:
- Add a property to the container runtime's metadata blob during container creation. The value can be read from the container runtime options and tells whether the feature is on / off.
- For all other container runtime that loads, use the property value from the metadata blob to turn the feature on / off. Ignore what the container runtime options says.

See `gcEnabled` in `gcConfigs.ts` for an example of this. The limitation for this is that you cannot enable this for older documents that were created before this feature was added and you cannot toggle this feature after a container is created. That's why my seconds question above :-).

Reply:
Nice, this is a pattern I hadn't seen before. I've pushed a commit that sets the property in the runtime's metadata and uses that to override any specified options.